src/coreclr/inc/corinfo.h

```diff
@@ -1684,6 +1684,7 @@ enum CORINFO_FIELD_ACCESSOR
     CORINFO_FIELD_STATIC_ADDR_HELPER,       // static field accessed using address-of helper (argument is FieldDesc *)
     CORINFO_FIELD_STATIC_TLS,               // unmanaged TLS access
     CORINFO_FIELD_STATIC_TLS_MANAGED,       // managed TLS access
+    CORINFO_FIELD_STATIC_TLS_MANAGED_LAZY,  // managed TLS access lazy ctor initialization
```

Comment:
Would it be better for this to be a flag instead of a new type of field accessor?

There is `CORINFO_FLG_FIELD_INITCLASS` that seems to be what we need.

Reply:
> There is `CORINFO_FLG_FIELD_INITCLASS` that seems to be what we need.

This flag solved lot of the problems.
- No need for `CORINFO_FIELD_STATIC_TLS_MANAGED_LAZY`
- No need to add the extra `GCStaticBase` helper call during importing
- No need of tracking `enclosingType` and using it in VN
- The `gtEntryPoint` gets setup to correct symbol
- Eliminates the extra thread static helper remaining discussed in https://github.com/dotnet/runtime/pull/89472#discussion_r1430540739